File: typescript/8kyu/test/even-or-odd.test.ts
import { evenOrOdd } from "../solution/even-or-odd";

type TestCase = { input: number; output: string };

const testCases: TestCase[] = [
  { input: 2, output: "Even" },
  { input: 7, output: "Odd" },
  { input: -42, output: "Even" },
  { input: -7, output: "Odd" },
  { input: 0, output: "Even" },
  { input: 1, output: "Odd" },
  { input: -1, output: "Odd" },
  { input: 100, output: "Even" },
  { input: -100, output: "Even" },
  { input: 1000, output: "Even" },
];

describe('Create a function that takes an integer as an argument and returns "Even" for even numbers or "Odd" for odd numbers.', () => {
  testCases.forEach(({ input, output }, index) => {
    it(`${index + 1}. Input: ${input}`, () => {
      expect(evenOrOdd(input)).toStrictEqual(output);
    });
  });
});

File: python/8kyu/test/even_or_odd_test.py
import pytest
from solution.even_or_odd import even_or_odd

def test_even_number():
    assert even_or_odd(2) == "Even"
    assert even_or_odd(100) == "Even"

def test_odd_number():
    assert even_or_odd(1) == "Odd"
    assert even_or_odd(99) == "Odd"

@pytest.mark.parametrize("number, expected", [
    (0, "Even"),
    (7, "Odd"),
    (-2, "Even"),
    (-5, "Odd"),
    (42, "Even"),
])
def test_even_or_odd_parametrized(number, expected):
    assert even_or_odd(number) == expected
